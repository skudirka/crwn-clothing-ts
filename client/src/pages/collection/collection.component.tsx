import React from 'react';
import {connect} from 'react-redux';

import CollectionItem from '../../components/collection-item/collection-item.component';

import {selectCollection} from '../../redux/shop/shop.selectors';

import {CollectionPageContainer, CollectionTitle, CollectionItemsContainer} from './collection.styles';

export const CollectionPage = ({collection}) => {
    const {title, items} = collection;
    return (
        <CollectionPageContainer className="collection-page">
            <CollectionTitle>{title}</CollectionTitle>
            <CollectionItemsContainer className="items">
                {
                    items.map(item => (
                        <CollectionItem key={item.id} item={item} className="collection-item" />
                    ))
                }
            </CollectionItemsContainer>
        </CollectionPageContainer>
    )
};

const mapStateToProps = (state, ownProps) => ({
    collection: selectCollection(ownProps.match.params.collectionId)(state)
})

export default connect(mapStateToProps)(React.memo(CollectionPage));