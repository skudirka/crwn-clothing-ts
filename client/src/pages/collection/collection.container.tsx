import {createStructuredSelector} from 'reselect';
import {connect, ConnectedProps} from 'react-redux';
import {compose} from 'redux';

import {selectIsCollectionsLoaded} from '../../redux/shop/shop.selectors';

import CollectionPage from './collection.component';
import WithSpinner from '../../components/with-spinner/with-spinner.component';
import {IRootState} from '../../redux/root-types';

interface mapStateToPropsInterface {
    isLoading:boolean;
}
const mapStateToProps = createStructuredSelector<IRootState, mapStateToPropsInterface>({
    isLoading: state => !selectIsCollectionsLoaded(state)
});

const connector = connect(
    mapStateToProps
);

type PropsFromRedux = ConnectedProps<typeof connector>;

type CollectionPageProps = PropsFromRedux & {};

const CollectionPageContainer = compose(
    connector,
    WithSpinner
)(CollectionPage) as React.FC<CollectionPageProps>;

export default CollectionPageContainer;