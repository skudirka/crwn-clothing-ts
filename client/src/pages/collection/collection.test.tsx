import {shallow, mount} from 'enzyme';
import React from 'react';

import {CollectionPage} from './collection.component';

describe('CollectionPage page', () => {
    let wrapper;
    let mockProps;

    const mockCollection = {
        id: '1',
        name: 'Collection1',
        items: [
            {
                id: '1',
                name: 'Item1',
                imageUrl: 'https://via.placeholder.com/150',
                price: 10,
                quantity: 0,
            },
            {
                id: '2',
                name: 'Item2',
                imageUrl: 'https://via.placeholder.com/150',
                price: 20,
                quantity: 0,
            },
            {
                id: '3',
                name: 'Item3',
                imageUrl: 'https://via.placeholder.com/150',
                price: 13,
                quantity: 0,
            },
            {
                id: '4',
                name: 'Item4',
                imageUrl: 'https://via.placeholder.com/150',
                price: 16,
                quantity: 0,
            },
            {
                id: '5',
                name: 'Item5',
                imageUrl: 'https://via.placeholder.com/150',
                price: 9,
                quantity: 0,
            }
        ]
    }

    beforeEach(() => {

        mockProps = {
            collection: mockCollection
        };

        wrapper = shallow(<CollectionPage {...mockProps} />);
    });

    it('should render CollectionPage page', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render same number of CollectionItem as the number of items in the collection', () => {
        expect(wrapper.find('.collection-item').length).toEqual( mockCollection.items.length );
    });
})