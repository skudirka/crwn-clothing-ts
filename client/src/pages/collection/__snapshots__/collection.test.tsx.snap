// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage page should render CollectionPage page 1`] = `ShallowWrapper {}`;
